Config.cmake.in

```diff
@@ -0,0 +1,11 @@
+@PACKAGE_INIT@
+
+include(CMakeFindDependencyMacro)
+
+find_dependency(${PROJ4_LIBRARIES} REQUIRED )
+find_dependency(units::units REQUIRED )
+find_dependency(GSL REQUIRED )
+
+
+check_required_components(@PROJECT_NAME@)
```

Comment:
Was reading the following documentation about this call [here](https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html). Is this unnecessary in your mind because the to `configure_package_config_file ()` in carma-builds InstallingConfigs.cmake should generate this macros by default as long as we do not specific `NO_CHECK_REQUIRED_COMPONENTS_MACRO` option?

Reply:
No, it is unnecessary because modern CMake uses target-based configurations, and the targets should be responsible for checking that their dependencies are satisfied. Older CMake used instead of targets, so it was necessary to check that the variables were all defined properly. Here's the relevant excerpt from Craig Scott's book:

> In the past when all details about a package were provided through variables, it was customary to check whether all required variables were set at the end of the config file before returning. A macro called `check_required_components()` was defined for this purpose, but projects that provide imported targets should perform these checks themselves... This makes the `check_required_componetns()` macro largely redundant.

Reply:
Personally I think this sounds more like a preference Craig Scott has in setting of CMake configurations to move responsibility for checking required components to the consuming project since the CMake official documentation still suggests we use this macros for CMake configs. I am not opposed to using a better approach as long as the benefit of the change is worth the effort involved. In this case, I think if we want to change our CMake library install configurations to match closer to Craig Scotts approach I suggest we do this in a separate PR and issue so we can prioritize this work against the other improvements we need to make and so that we can implement this change more consistently across projects